### PR TITLE
Add switch todo status feature and process marks for selected lines

### PR DESCRIPTION
**Pull request by xianzhon**
Mainly solve issue `#21 Mark all selected lines as TODO items or if it is, mark as DONE`

### CODE REVIEW

**Comment by vitalk** [comment]
Your PR looks great! I can test it a little bit later. You can fix some style issues for consistency.

File: plugin/simple-todo.vim [hunk 1]
```diff
@@ -88,6 +88,57 @@ vnore <silent> <Plug>(simple-todo-mark-as-undone) :execute 's/^\(\s*[-+*]\?\s*\)
...
+    let [lnum1, lnum2] = [a:1, a:2]
+  else
+    let [lnum1, lnum2] = [line("'["), line("']")]
+  endif
+
+  let s:mark_todo = '\1[ ]'
+  let s:mark_done = '\1[' . g:simple_todo_tick_symbol . ']'
+  let s:pat_mark_todo = '^\(\s*[-+*]\?\s*\)\[ \]'
+  let s:pat_mark_done = '^\(\s*[-+*]\?\s*\)\[' . g:simple_todo_tick_symbol . ']'
+  for lnum in range(lnum1,lnum2)
```
Comment (vitalk) [inline plugin/simple-todo.vim]: Missed space after `,`

File: plugin/simple-todo.vim [hunk 2]
```diff
@@ -88,6 +88,57 @@ vnore <silent> <Plug>(simple-todo-mark-as-undone) :execute 's/^\(\s*[-+*]\?\s*\)
...
+    let line = getline(lnum)
+    if strlen(line) > 0
+        if a:type == 0 " add [ ]
+            if line !~ s:pat_mark_todo && line !~ s:pat_mark_done
+              let line = substitute(line,'^\(\s*\)',s:mark_todo.' ','')
```
Comment (vitalk) [inline plugin/simple-todo.vim]: Same as above

File: plugin/simple-todo.vim [hunk 3]
```diff
@@ -88,6 +88,57 @@ vnore <silent> <Plug>(simple-todo-mark-as-undone) :execute 's/^\(\s*[-+*]\?\s*\)
...
+            endif
+        elseif a:type == 1 " mark done
+            if line =~ s:pat_mark_todo
+              let line = substitute(line,s:pat_mark_todo, s:mark_done ,'')
```
Comment (vitalk) [inline plugin/simple-todo.vim]: Same as above

File: plugin/simple-todo.vim [hunk 4]
```diff
@@ -88,6 +88,57 @@ vnore <silent> <Plug>(simple-todo-mark-as-undone) :execute 's/^\(\s*[-+*]\?\s*\)
...
+            endif
+        elseif a:type == 2 " mark undone
+            if line =~ s:pat_mark_done
+              let line = substitute(line,s:pat_mark_done,s:mark_todo,'')
```
Comment (vitalk) [inline plugin/simple-todo.vim]: Same as above

File: plugin/simple-todo.vim [hunk 5]
```diff
@@ -88,6 +88,57 @@ vnore <silent> <Plug>(simple-todo-mark-as-undone) :execute 's/^\(\s*[-+*]\?\s*\)
...
+            endif
+        elseif a:type == 3 " switch mark
+            if line =~ s:pat_mark_todo
+              let line = substitute(line,s:pat_mark_todo,s:mark_done ,'')
```
Comment (vitalk) [inline plugin/simple-todo.vim]: Same as above

File: plugin/simple-todo.vim [hunk 6]
```diff
@@ -88,6 +88,57 @@ vnore <silent> <Plug>(simple-todo-mark-as-undone) :execute 's/^\(\s*[-+*]\?\s*\)
...
+            elseif line =~ s:pat_mark_done
+              let line = substitute(line,s:pat_mark_done,s:mark_todo,'')
```
Comment (vitalk) [inline plugin/simple-todo.vim]: Same as above

File: plugin/simple-todo.vim [hunk 7]
```diff
@@ -88,6 +88,57 @@ vnore <silent> <Plug>(simple-todo-mark-as-undone) :execute 's/^\(\s*[-+*]\?\s*\)
...
+            endif
+        endif
+        call setline(lnum,line)
```
Comment (vitalk) [inline plugin/simple-todo.vim]: Same as above

File: plugin/simple-todo.vim [hunk 8]
```diff
@@ -88,6 +88,57 @@ vnore <silent> <Plug>(simple-todo-mark-as-undone) :execute 's/^\(\s*[-+*]\?\s*\)
 inore <silent> <Plug>(simple-todo-mark-as-undone) <Esc>:execute 's/^\(\s*[-+*]\?\s*\)\[' . g:simple_todo_tick_symbol . ']/\1[ ]/'<cr>
       \:silent! call repeat#set("\<Plug>(simple-todo-mark-as-undone)")<cr>
 
+function! s:go(type,...) abort
```
Comment (vitalk) [inline plugin/simple-todo.vim]: Move out `s:go` definiton from `Public API` section, for consistency.

File: doc/simple-todo.txt [hunk 9]
```diff
@@ -66,6 +66,10 @@ or disabled separately for each mode
 <leader>X                   Mark item under cursor as undone
 
                                            *<Plug>(simple-todo-new-list-item)*
+<Plug>(simple-todo-mark-switch)
+<leader>s                   Switch todo status for current line
+
+                                             *<Plug>(simple-todo-mark-switch)*
```
Comment (vitalk) [inline doc/simple-todo.txt]: @xianzhon, one more last thing. You add docstring for wrong line, should be

```
                                             *<Plug>(simple-todo-mark-switch)*
<Plug>(simple-todo-mark-switch)
<leader>s                   Switch todo status for current line
```

File: plugin/simple-todo.vim [hunk 10]
```diff
@@ -88,6 +126,22 @@ vnore <silent> <Plug>(simple-todo-mark-as-undone) :execute 's/^\(\s*[-+*]\?\s*\)
 inore <silent> <Plug>(simple-todo-mark-as-undone) <Esc>:execute 's/^\(\s*[-+*]\?\s*\)\[' . g:simple_todo_tick_symbol . ']/\1[ ]/'<cr>
       \:silent! call repeat#set("\<Plug>(simple-todo-mark-as-undone)")<cr>
 
+" Switch marks for visual selected lines
+nnoremap <silent> <Plug>(simple-todo-mark-switch)           :<C-U>call <SID>go(3, line("."), line("."))<CR>
+        \:silent! call repeat#set("\<Plug>(simple-todo-mark-switch)")<cr>
+inoremap <silent> <Plug>(simple-todo-mark-switch)           :<C-U>call <SID>go(3, line("."), line("."))<CR>
```
Comment (vitalk) [inline plugin/simple-todo.vim]: Go to normal mode before `:call`, e.g. `<Esc>:<C-U>call`

Comment (xianzhon) [inline plugin/simple-todo.vim]: Thank you for pointing out, and sorry for not testing thoroughly.

**Comment by vitalk** [comment]
Great! Thank you @xianzhon!

**Pull request merged**